### PR TITLE
fbjni: init at 0.3.0

### PR DESCRIPTION
**Pull request by jonringer**
###### Description of changes
Had to use this recently, and thought it would be useful for some people.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 182286` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fbjni</li>
  </ul>
</details>

File: pkgs/development/libraries/fbjni/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, fetchpatch
+, openjdk
```
Comment (azahi) [inline pkgs/development/libraries/fbjni/default.nix:6]: A nitpick.
```suggestion
, jdk
```

File: pkgs/development/libraries/fbjni/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  src = fetchFromGitHub {
+    owner = "facebookincubator";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "132s9c8vlx1r4zcjjsgii9n3f2fx5qxgj3yvg6hw0dzg61x07bpg";
+  };
+
+  patches = [
+    # Part of https://github.com/facebookincubator/fbjni/pull/76
+    # fix cmake file instllation directory
```
Comment (azahi) [inline pkgs/development/libraries/fbjni/default.nix:23]:
```suggestion
    # fix cmake file installation directory
```

**Review by azahi — DISMISSED** [review]

File: pkgs/development/libraries/fbjni/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/libraries/fbjni/default.nix:6]: These seem to be aliased, and i would rather use openjdk

```
[17:23:03] jon@jon-desktop /home/jon/projects/nixpkgs (arcanist-php80)
$ nix-build -A openjdk
/nix/store/m2gxaicfdpfbxpxs5j0j0qrih4m677y1-openjdk-17.0.3+7
[17:23:15] jon@jon-desktop /home/jon/projects/nixpkgs (arcanist-php80)
$ nix-build -A jdk
/nix/store/m2gxaicfdpfbxpxs5j0j0qrih4m677y1-openjdk-17.0.3+7
```

**Pull request merged**